tests/acceptance/course-page/code-examples/export-to-github-test.js

```diff
@@ -0,0 +1,173 @@
+    await percySnapshot('Code Examples - View on GitHub button (with existing export)');
+
+    await codeExamplesPage.solutionCards[0].highlightedFileCards[0].clickOnViewOnGithubButton();
+    await settled();
+
+    assert.notOk(exportCreateCalled, 'should reuse existing export');
+  });
+
+  test('creates new export when existing export is expired', async function (assert) {
+    const { solution } = await setupTestScenario.call(this, this.server);
+
+    this.server.create('community-solution-export', {
+      status: 'provisioned',
+      expiresAt: new Date(Date.now() - 25 * 60 * 60 * 1000),
+      githubRepositoryUrl: 'https://github.com/expired-repo/test',
+      communitySolution: solution,
+    });
+
+    let exportCreateCalled = false;
+    this.server.post('/community-solution-exports', function (schema) {
+      exportCreateCalled = true;
+
+      return schema.communitySolutionExports.create({
+        status: 'provisioned',
+        expiresAt: new Date(Date.now() + 24 * 60 * 60 * 1000),
+        githubRepositoryUrl: 'https://github.com/new-repo/test',
+        communitySolution: solution,
+      });
+    });
+
+    await navigateToCodeExamples();
+    await codeExamplesPage.solutionCards[0].highlightedFileCards[0].clickOnViewOnGithubButton();
+    await settled();
+
+    assert.ok(exportCreateCalled, 'should create new export when existing is expired');
+  });
+
+  test('shows direct GitHub link when solution is published', async function (assert) {
+    const { solution } = await setupTestScenario.call(this, this.server);
+
+    solution.update({
+      githubRepositoryName: 'author/redis-solution',
+      githubRepositoryIsPrivate: false,
+      commitSha: 'abc123',
+    });
+
+    let exportCreateCalled = false;
+    this.server.post('/community-solution-exports', function () {
+      exportCreateCalled = true;
+
+      return { data: {} };
+    });
+
+    await navigateToCodeExamples();
+    await percySnapshot('Code Examples - Direct GitHub link (published solution)');
+
+    // Published solutions show a direct link, not a button
+    assert.dom('[data-test-view-on-github-button]').doesNotExist('should not show view button for published solutions');
+    assert.dom('a[href*="github.com/author/redis-solution"]').exists('should show direct GitHub link');
+    assert.notOk(exportCreateCalled, 'should not create export when solution is published');
+  });
+
+  test('handles export creation failure gracefully', async function (assert) {
+    await setupTestScenario.call(this, this.server);
+
+    this.server.post('/community-solution-exports', function () {
+      return new Response(500, {}, { error: 'Export creation failed' });
+    });
+
+    await navigateToCodeExamples();
+
+    await codeExamplesPage.solutionCards[0].highlightedFileCards[0].clickOnViewOnGithubButton();
+    await settled();
+
+    assert.dom('[data-test-view-on-github-button]').exists('view button should still exist after failure');
```

Comment:
Use page objects, not `assert.dom`. We use `assert.dom` in very few places and only when the use-case can't be achieved via page objects.

Reply:
This is another example where just searching the codebase will give you an idea of what patterns we follow. Look for `assert.dom` usages and you'll find very few